pkg/ccl/changefeedccl/kvevent/alloc.go

```diff
@@ -36,14 +45,30 @@ func (a *Alloc) Merge(other *Alloc) {
 	}
 
 	if a.ap != other.ap {
-		panic("cannot merge allocations from two different pools")
+		// Slow case: this doesn't happen frequently (only right after backfill completes).
+		a.mergeSlow(other)
+		return
 	}
+
 	a.bytes += other.bytes
 	a.entries += other.entries
 	other.bytes = 0
 	other.entries = 0
 }
 
+// mergeSlow merges allocation that belongs to another alloc pool.
+func (a *Alloc) mergeSlow(other *Alloc) {
+	if a.otherPoolAllocs == nil {
+		a.otherPoolAllocs = make(map[int64]*Alloc, 1)
+	}
+	id := *(*int64)(unsafe.Pointer(&a.ap))
```

Comment:
this is going to be different for every `Alloc`. You need to either type assert to get a pointer or use the interface type (`pool`) in the map. My preference is for the latter.